packages/material-ui/src/Tabs/ScrollbarSize.js

```diff
@@ -18,53 +18,49 @@ const styles = {
  * The component is originates from https://github.com/STORIS/react-scrollbar-size.
  * It has been moved into the core in order to minimize the bundle size.
  */
-class ScrollbarSize extends React.Component {
-  constructor() {
-    super();
+function ScrollbarSize(props) {
+  const { onChange } = props;
+  const scrollbarHeight = React.useRef();
+  const nodeRef = React.useRef();
 
-    if (typeof window !== 'undefined') {
-      this.handleResize = debounce(() => {
-        const prevHeight = this.scrollbarHeight;
-        this.setMeasurements();
-
-        if (prevHeight !== this.scrollbarHeight) {
-          this.props.onChange(this.scrollbarHeight);
-        }
-      }, 166); // Corresponds to 10 frames at 60 Hz.
+  const setMeasurements = () => {
+    if (!nodeRef.current) {
+      return;
     }
-  }
 
-  componentDidMount() {
-    this.setMeasurements();
-    this.props.onChange(this.scrollbarHeight);
-  }
+    scrollbarHeight.current = nodeRef.current.offsetHeight - nodeRef.current.clientHeight;
+  };
+
+  let handleResize = () => {};
 
-  componentWillUnmount() {
-    this.handleResize.clear();
+  if (typeof window !== 'undefined') {
+    handleResize = debounce(() => {
+      const prevHeight = scrollbarHeight.current;
+      setMeasurements();
+
+      if (prevHeight !== scrollbarHeight.current) {
+        onChange(scrollbarHeight.current);
+      }
+    }, 166); // Corresponds to 10 frames at 60 Hz.
   }
 
-  handleRef = ref => {
-    this.nodeRef = ref;
+  const handleRef = ref => {
+    nodeRef.current = ref;
   };
 
-  setMeasurements = () => {
-    const nodeRef = this.nodeRef;
-
-    if (!nodeRef) {
-      return;
-    }
+  React.useEffect(() => {
+    setMeasurements();
+    onChange(scrollbarHeight.current);
 
-    this.scrollbarHeight = nodeRef.offsetHeight - nodeRef.clientHeight;
-  };
+    return () => handleResize.clear();
```

Comment:
Something is off. A new `handleResize` function is created at each render, and attached to the event listener, but the clear method is called on a scoped variable, if the onChange method doesn't change and the component rerender, the clear() will be applied on an outdated event handler.

Reply:
I would suggest to kill EventListener, do everything in the useEffect.

Reply:
Hooks make the so much cleaner!